src/domain/errors/Errors.ts

```diff
@@ -3,3 +3,22 @@ export type ValidationError<T> = {
     value: unknown;
     errors: string[];
 };
+
+export function isValidationErrorArray<T>(value: unknown): value is ValidationError<T>[] {
+    return (
```

Comment:
Ideally, in a point of domain where we have full control we should need to handle unknowns and runtime checks, but again, this is related to the Either method called from everywhere, so we can keep it as-is.